src/plugins/kibana/server/routes/api/ingest/register_bulk.js

```diff
@@ -0,0 +1,97 @@
+import { Promise } from 'bluebird';
+import { parse, transform } from 'csv';
+import _ from 'lodash';
+import hi from 'highland';
+import { patternToIngest } from '../../../../common/lib/convert_pattern_and_ingest_name';
+import { PassThrough } from 'stream';
+import bulkRequestSchema from '../../../lib/schemas/bulk_request_schema';
+import JSONStream from 'JSONStream';
+
+export function registerBulk(server) {
+  server.route({
+    path: '/api/kibana/{id}/_bulk',
```

Comment:
I think we should stay away from `_bulk` as it has a certain connotation in Elasticsearch. That said, I don't think we should have separate endpoints for different _formats_ of data; that's what the `Content-Type` request header is for. As this API exists in service of the add data UI, what about calling the endpoint `_data`?